scalding-core/src/main/scala/com/twitter/scalding/JobClassFinder.scala

```diff
@@ -0,0 +1,60 @@
+package com.twitter.scalding
+
+import com.twitter.scalding.typed.CoGroupable
+
+import scala.reflect.runtime.universe
+
+object JobClassFinder {
```

Comment:
I don't think `Job` has anything really to do with this. Can we just say `ReferencedClassFinder` or something? Also, we might add a method like:
```scala
object ReferencedClassFinder {
  /**
   * Add the given type, as well as all referenced types to the cascading tokens list.
   * note, for maximal efficiency, you should also register those types with the kryo
   * instantiator being used.
   */
  def addCascadingTokensFrom(c: Class[_], config: Config): Config
}
```